packages/blocks/src/root-block/edgeless/edgeless-keyboard.ts

```diff
@@ -86,6 +87,7 @@ export class EdgelessPageKeyboardManager extends PageKeyboardManager {
           });
         },
         m: () => {
+          if (this.rootElement.service.tool.locked) return;
```

Comment:
Checking for a lock each time seems a bit inelegant. I'm wondering if we could add a lock attribute to the viewport class, allowing calls to fail silently if it's locked.

Reply:
I moved it to `service`.